src/omemo/QXmppOmemoManager_p.cpp

```diff
@@ -0,0 +1,3714 @@
+        if (const auto error = std::get_if<Error>(&result)) {
+            warning("Features of PEP service '" % ownBareJid() % "' could not be retrieved" % errorToString(*error));
+            warning("Device bundle and device list could not be published");
+            reportFinishedResult(interface, false);
+        } else {
+            const auto &pepServiceFeatures = std::get<QVector<QString>>(result);
+
+            // Check if the PEP service supports publishing items at all and also publishing
+            // multiple items.
+            // The support for publishing multiple items is needed to publish multiple device
+            // bundles to the corresponding node.
+            // It is checked here because if that is not possible, the publication of the device
+            // element must not be published.
+            // TODO: Uncomment the following line and remove the other one once ejabberd released version > 21.12
+            // if (pepServiceFeatures.contains(ns_pubsub_publish) && pepServiceFeatures.contains(ns_pubsub_multi_items)) {
+            if (pepServiceFeatures.contains(ns_pubsub_publish)) {
+                auto future = pubSubManager->fetchPepNodes();
+                await(future, q, [=](QXmppPubSubManager::NodesResult result) mutable {
+                    if (const auto error = std::get_if<Error>(&result)) {
+                        warning("Nodes of JID '" % ownBareJid() % "' could not be fetched to check if nodes '" %
+                                QString(ns_omemo_2_bundles) % "' and '" % QString(ns_omemo_2_devices) %
+                                "' exist" % errorToString(*error));
+                        warning("Device bundle and device list could not be published");
+                        reportFinishedResult(interface, false);
+                    } else {
+                        const auto &nodes = std::get<QVector<QString>>(result);
+
+                        const auto deviceListNodeExists = nodes.contains(ns_omemo_2_devices);
+                        const auto arePublishOptionsSupported = pepServiceFeatures.contains(ns_pubsub_publish_options);
+                        const auto isAutomaticCreationSupported = pepServiceFeatures.contains(ns_pubsub_auto_create);
+                        const auto isCreationAndConfigurationSupported = pepServiceFeatures.contains(ns_pubsub_create_and_configure);
+                        const auto isCreationSupported = pepServiceFeatures.contains(ns_pubsub_create_nodes);
+                        const auto isConfigurationSupported = pepServiceFeatures.contains(ns_pubsub_config_node);
+
+                        // The device bundle is published before the device data is published.
+                        // That way, it ensures that other devices are notified about this new
+                        // device only after the corresponding device bundle is published.
+                        auto handleResult = [=, this](bool isPublished) mutable {
```

Comment:
```suggestion
                        auto handleResult = [=](bool isPublished) mutable {
```

Reply:
using `[=]` instead of `[=, this]` results in a warning with C++20

https://stackoverflow.com/questions/73196817/why-do-lambda-functions-need-to-capture-this-pointer-explicitly-in-c20